### DISCUSSION TITLE
How to enable `trim-diagnostic-paths` in clippy?

### DISCUSSION THREAD

**Discussion by GordianDziwis** [rust-lang/rust-clippy#11884] [Q&A] [answered]
With `cargo build` I get:

```
error[E0277]: the trait bound `Rc<str>: From<&std::string::String>` is not satisfied
```

and with `cargo clippy`:
```
error[E0277]: the trait bound `std::rc::Rc<str>: std::convert::From<&std::string::String>` is not satisfied
```

This does not help:
```
RUSTFLAGS="-Z trim-diagnostic-paths=true" cargo  clippy
```

Is this by design?

**Comment by flip1995** [comment, marked as answer]
We were having a bunch of issues with this feature and decided not to enable it when it was implemented in `rustc`. See: https://github.com/rust-lang/rust-clippy/issues/7798 (and the linked issues + PR for details and long discussions)

The situation might have changed since we last attempted this. So looking into this might be an interesting project.
